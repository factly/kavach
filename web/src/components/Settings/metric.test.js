import React from 'react';
import { mount } from 'enzyme';

import Metric from './metric';

describe('Metric', () => {
  it('should render and match snapshot with props passed', () => {
    const wrapper = mount(<Metric count={1} header="Test" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render and match snapshot without optional props passed', () => {
    const wrapper = mount(<Metric header={'test'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
