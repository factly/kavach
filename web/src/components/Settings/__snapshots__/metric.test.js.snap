// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metric should render and match snapshot with props passed 1`] = `
<Metric
  count={1}
  header="Test"
>
  <div
    style={
      Object {
        "alignItems": "flex-start",
        "background": "#fff",
        "border": "1px solid #eaecf0",
        "borderRadius": "8px",
        "boxShadow": "0px 1px 3px rgba(16, 24, 40, 0.1), 0px 1px 2px rgba(16, 24, 40, 0.06)",
        "display": "flex",
        "flexDirection": "column",
        "gap": "1rem",
        "height": "120px",
        "isolation": "isolate",
        "maxWidth": "180px",
        "padding": "1rem",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "color": "#101828",
          "flex": "none",
          "flexGrow": 1,
          "fontSize": "16px",
          "fontWeight": "500",
          "height": "24px",
          "lineHeight": "24px",
          "order": 1,
        }
      }
    >
      Test
    </div>
    <div
      style={
        Object {
          "alignSelf": "stretch",
          "color": "#101828",
          "flex": "none",
          "flexGrow": 0,
          "fontSize": "36px",
          "fontWeight": "600",
          "height": "44px",
          "letterSpacing": "-0.02em",
          "lineHeight": "44px",
          "order": 0,
        }
      }
    >
      1
    </div>
  </div>
</Metric>
`;

exports[`Metric should render and match snapshot without optional props passed 1`] = `
<Metric
  header="test"
>
  <div
    style={
      Object {
        "alignItems": "flex-start",
        "background": "#fff",
        "border": "1px solid #eaecf0",
        "borderRadius": "8px",
        "boxShadow": "0px 1px 3px rgba(16, 24, 40, 0.1), 0px 1px 2px rgba(16, 24, 40, 0.06)",
        "display": "flex",
        "flexDirection": "column",
        "gap": "1rem",
        "height": "120px",
        "isolation": "isolate",
        "maxWidth": "180px",
        "padding": "1rem",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "color": "#101828",
          "flex": "none",
          "flexGrow": 1,
          "fontSize": "16px",
          "fontWeight": "500",
          "height": "24px",
          "lineHeight": "24px",
          "order": 1,
        }
      }
    >
      test
    </div>
    <div
      style={
        Object {
          "alignSelf": "stretch",
          "color": "#101828",
          "flex": "none",
          "flexGrow": 0,
          "fontSize": "36px",
          "fontWeight": "600",
          "height": "44px",
          "letterSpacing": "-0.02em",
          "lineHeight": "44px",
          "order": 0,
        }
      }
    >
      --
    </div>
  </div>
</Metric>
`;
